lib/active_admin/filters/resource_extension.rb

```diff
@@ -174,8 +174,21 @@ def search_status_section
                 else
                   active.filters.each do |filter|
                     li do
-                      span filter.body
-                      b filter.value
+                      if filter.body =~ /ID/i
+                        begin
+                          klass = filter.body.split(' ').first.constantize
+                          method = klass.inspect.scan(/(title|name|value|id)/).last[0].to_sym
+                          value = klass.find(filter.value).send(method)
```

Comment:
@ccallebs 
You're right with this.
I didn't want to change other code a lot, thought this monkey-patch can help for the first time ))